clouddriver-core/src/main/java/com/netflix/spinnaker/clouddriver/config/AccountDefinitionConfiguration.java

```diff
@@ -63,110 +68,112 @@
 @Configuration
 @EnableConfigurationProperties(AccountDefinitionConfiguration.Properties.class)
 @Log4j2
+@RequiredArgsConstructor
 public class AccountDefinitionConfiguration {
 
+  private final Properties properties;
+
   @Bean
   @ConditionalOnMissingBean
-  public AccountDefinitionAuthorizer accountDefinitionAuthorizer(
-      Optional<FiatPermissionEvaluator> maybePermissionEvaluator,
-      @Value("${services.fiat.enabled:false}") boolean isFiatEnabled) {
-    return new AccountDefinitionAuthorizer(
-        maybePermissionEvaluator.filter(ignored -> isFiatEnabled).orElse(null));
+  public AccountSecurityPolicy accountSecurity(
+      @Nullable FiatPermissionEvaluator permissionEvaluator,
+      @Value("${services.fiat.enabled:false}") boolean fiatEnabled) {
+    return fiatEnabled && permissionEvaluator != null
```

Comment:
Trying to wrap my head around when FiatPermissionEvaluator could be null...[CacheConfig](https://github.com/spinnaker/clouddriver/blob/0f1c3b83d2e8673bf4d545ab6f72445dfea988cb/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CacheConfig.groovy#L111) has it Optional which seems like a good precedent, though now that I've dug I don't think that's necessary.  It's a long and winding path, but..

* FiatPermissionEvaluator is annotated with [`@Component`](https://github.com/spinnaker/fiat/blob/8b6d86cea9ca40df3476f60a71dc1988c93b8ff6/fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java#L58) and in a package (`com.netflix.spinnaker.fiat.shared`) that's `@ComponentScan`ned by [FiatAuthenticationConfig](https://github.com/spinnaker/fiat/blob/8b6d86cea9ca40df3476f60a71dc1988c93b8ff6/fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAuthenticationConfig.java#L55).
* FiatAuthenticationConfig comes in via [EnableFiatAutoConfig](https://github.com/spinnaker/fiat/blob/8b6d86cea9ca40df3476f60a71dc1988c93b8ff6/fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/EnableFiatAutoConfig.java#L43).
* clouddriver's SecurityConfig has [`@EnableFiatAutoConfig`](https://github.com/spinnaker/clouddriver/blob/0f1c3b83d2e8673bf4d545ab6f72445dfea988cb/clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/config/SecurityConfig.groovy#L31).
* clouddriver's Main imports [SecurityConfig](https://github.com/spinnaker/clouddriver/blob/0f1c3b83d2e8673bf4d545ab6f72445dfea988cb/clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/Main.groovy#L46).

Reply:
IntelliJ complains about this a bit, though yeah, this is typically available at least even if Fiat is disabled since that component short circuits its APIs when Fiat is disabled. I think I used `@Nullable` here instead of `Optional` as, again, IntelliJ static code analysis complains about using `Optional` in parameters and fields, and given our annotation processing, you get the same benefits from using `@Nullable` as `Optional` in such use cases (and Spring supports both).

I think this could be refactored a little to work more like `FiatPermissionEvaluator` where it checks `FiatStatus` for whether or not Fiat is enabled rather than a startup configuration bean.

Reply:
Optional vs. Nullable isn't so interesting for me.  I did try taking it out locally and ran into a bunch of struggles providing the right beans in some tests.  You may have an easier time with it than me.  It's also OK to punt on this for this PR if it's too painful.